Signal/src/call/CallService.swift

```diff
@@ -1198,34 +1163,35 @@ protocol CallServiceObserver: class {
     // that we want to terminate the current call (if any) in order to
     // return to a known good state.
     public func handleFailedCurrentCall(error: CallError) {
-        handleFailedCall(failedCall: self.call, error: error, forceTerminate:true)
+        Logger.debug("\(TAG) in \(#function)")
+
+        // Return to a known good state by ending the current call, if any.
+        handleFailedCall(failedCall: self.call, error: error)
     }
 
     // This method should be called when a fatal error occurred for a call.
     //
     // * If we know which call it was, we should update that call's state
     //   to reflect the error.
     // * IFF that call is the current call, we want to terminate it.
-    public func handleFailedCall(failedCall: SignalCall?, error: CallError, forceTerminate: Bool = false) {
+    public func handleFailedCall(failedCall: SignalCall?, error: CallError) {
```

Comment:
I've modified `handleFailedCall()` to always `terminateCall()` when the failed call is nil (i.e. the `handleFailedCurrentCall()` case), which it wasn't before.  That was a bug.